File: lab6/homework/Carrier/src/main/kotlin/AdminNotificationsHandler.kt
import com.rabbitmq.client.Channel
import com.rabbitmq.client.DeliverCallback

class AdminNotificationsHandler(
    private val queueName: String,
    private val channel: Channel
) {
    fun handleNotifications() {
        val deliverCallback =
            DeliverCallback { _: String?, delivery ->
                val message = String(delivery.body, Charsets.UTF_8)
                println(" [x] ADMIN NOTIFICATION: $message")
            }

        channel.basicConsume(
            queueName,
            true,
            deliverCallback
        ) { consumerTag: String? ->
            println(" [x] Cancelled '$consumerTag'")
        }
    }
}

File: lab6/homework/Agency/AdminNotificationsHandler.cs
using System.Text;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;

namespace Agency;

public class AdminNotificationsHandler
{
    private readonly string _queueName;
    private readonly IModel _channel;

    public AdminNotificationsHandler(string queueName, IModel channel)
    {
        _queueName = queueName;
        _channel = channel;
    }

    public void HandleNotifications()
    {
        var consumer = new EventingBasicConsumer(_channel);
        consumer.Received += (_, ea) =>
        {
            var body = ea.Body.ToArray();
            var message = Encoding.UTF8.GetString(body);
            var routingKey = ea.RoutingKey;
            Console.WriteLine($"\b\b\b\b [x] ADMIN NOTIFICATION: {message}");
            Console.Write(">>> ");
        };
        
        _channel.BasicConsume(
            queue: _queueName,
            autoAck: true,
            consumer: consumer
        );
        
        Console.WriteLine("[*] Started listening for admin notifications");
    }
}